src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2168,6 +2168,44 @@ TEST_CASE("BilinearInterpolationLayerParametersTest", "[ANNLayerTest]")
   REQUIRE(layer1.InDepth() == layer2.InDepth());
 }
 
+/*
+ * Simple test for the BicubicInterpolation layer
+ */
+TEST_CASE("SimpleBicubicInterpolationLayerTest", "[ANNLayerTest]")
+{
+  // Tested output against  torch.nn.Upsample(mode="bicubic")
+  arma::mat input, output, unzoomedOutput, expectedOutput;
+  size_t inRowSize = 2;
+  size_t inColSize = 2;
+  size_t outRowSize = 5;
+  size_t outColSize = 7;
+  size_t depth = 1;
+  input.zeros(inRowSize * inColSize * depth, 1);
+
+  input << 10 << 20 << arma::endr
+        << 30 << 40 << arma::endr;
+  input.reshape(4, 1);
+  BicubicInterpolation<> layer(inRowSize, inColSize, outRowSize, outColSize, depth);
+
+  expectedOutput << 6.6880  <<  7.3331  <<   9.6973 <<  12.7950 <<  15.8927 <<  18.2569 << 18.9020 << arma::endr
+                 << 10.5330 <<  11.1781 <<  13.5423 <<  16.6400 <<  19.7377 <<  22.1019 << 22.7470 << arma::endr
+                 << 18.8930 <<  19.5381 <<  21.9023 <<  25.0000 <<  28.0977 <<  30.4619 << 31.1070 << arma::endr
+                 << 27.2530 <<  27.8981 <<  30.2623 <<  33.3600 <<  36.4577 <<  38.8219 << 39.4670 << arma::endr
+                 << 31.0980 <<  31.7431 <<  34.1073 <<  37.2050 <<  40.3027 <<  42.6669 << 43.3120 << arma::endr;
+  expectedOutput.reshape(35, 1);
+  layer.Forward(input, output);
+  CheckMatrices(output - expectedOutput, arma::zeros(output.n_rows), 1e-4);
+
+  expectedOutput.clear();
+  expectedOutput << 103.79045868 << 180.51335144 << arma::endr
+                 << 256.98651123 << 333.70950317 << arma::endr;
+  expectedOutput.reshape(4, 1);
+
+  layer.Backward(output, output, unzoomedOutput);
+  CheckMatrices(unzoomedOutput - expectedOutput, arma::zeros(input.n_rows), 1e-4);
```

Comment:
@zoq  hey, there is some precision error. Shall I reduce the precision to 1e-2 or 1e-3

Reply:
We can do that, can you also add a print statement at the end, would be interesting to see the output on the CI.

Reply:
Hey, you can take a look at this notebook 
https://colab.research.google.com/drive/1dqjdHB4-Rw1HNDd59gFFA468WEPY7oTf?usp=sharing
I have implemented the same in both pythorch and armadillo c++.
The output is same till 4th decimal digit, then it slightly varies.

Reply:
Ahh, because you don't compare against the full resolution, instead of decreasing the threshold why not use:

```
tensor([[[[ 6.6880393586,  7.3330830904,  9.6973323615, 12.7950000000,
           15.8926676385, 18.2569169096, 18.9019606414],
          [10.5330393586, 11.1780830904, 13.5423323615, 16.6400000000,
           19.7376676385, 22.1019169096, 22.7469606414],
          [18.8930393586, 19.5380830904, 21.9023323615, 25.0000000000,
           28.0976676385, 30.4619169096, 31.1069606414],
          [27.2530393586, 27.8980830904, 30.2623323615, 33.3600000000,
           36.4576676385, 38.8219169096, 39.4669606414],
          [31.0980393586, 31.7430830904, 34.1073323615, 37.2050000000,
           40.3026676385, 42.6669169096, 43.3119606414]]]],
```

I don't mind to drop the threshold.

Reply:
I have done this for now.   But my concern is the the output from my implementation differs slightly from the pytorch implementation.